build.sbt

```diff
@@ -294,40 +300,46 @@ lazy val enumeratumCirce = crossProject(JSPlatform, JVMPlatform)
   .in(file("enumeratum-circe"))
   .settings(commonWithPublishSettings: _*)
   .settings(testSettings: _*)
+  .jsSettings(jsTestSettings: _*)
   .settings(
     name := "enumeratum-circe",
     version := "1.5.24-SNAPSHOT",
     crossScalaVersions := scalaVersionsAll,
     libraryDependencies ++= {
       Seq(
-        "com.beachape" %%% "enumeratum" % Versions.Core.stable,
+        "com.beachape" %%% "enumeratum" % Versions.Core.head,
         "io.circe"     %%% "circe-core" % theCirceVersion(scalaVersion.value)
       )
     }
   )
+  .jsSettings(
+    crossScalaVersions := Seq(scala_2_12Version, scala_2_13Version)
+  )
+
 lazy val enumeratumCirceJs  = enumeratumCirce.js
 lazy val enumeratumCirceJvm = enumeratumCirce.jvm
 
 lazy val argonautAggregate =
-  aggregateProject("argonaut", enumeratumArgonautJs, enumeratumArgonautJvm)
+  aggregateProject("argonaut", /*enumeratumArgonautJs,*/ enumeratumArgonautJvm) // TODO re-enable once argonaut supports Scala.js 1.0
 lazy val enumeratumArgonaut = crossProject(JSPlatform, JVMPlatform)
   .crossType(CrossType.Pure)
   .in(file("enumeratum-argonaut"))
   .settings(commonWithPublishSettings: _*)
   .settings(testSettings: _*)
+  .jsSettings(jsTestSettings: _*)
   .settings(
     name := "enumeratum-argonaut",
     version := "1.5.14-SNAPSHOT",
     crossScalaVersions := scalaVersionsAll,
     libraryDependencies ++= {
       Seq(
-        "com.beachape" %%% "enumeratum" % Versions.Core.stable,
```

Comment:
The build is failing with quite a few errors, likely due to these changes; does `head` not resolve?

```
[error] sbt.librarymanagement.ResolveException: Error downloading com.beachape:enumeratum_2.12:1.5.16-SNAPSHOT
301[error]   Not found
302[error]   Not found
303[error]   not found: /home/travis/.ivy2/local/com.beachape/enumeratum_2.12/1.5.16-SNAPSHOT/ivys/ivy.xml
304[error]   not found: https://repo1.maven.org/maven2/com/beachape/enumeratum_2.12/1.5.16-SNAPSHOT/enumeratum_2.12-1.5.16-SNAPSHOT.pom
305[error]   not found: https://oss.sonatype.org/content/repositories/snapshots/com/beachape/enumeratum_2.12/1.5.16-SNAPSHOT/enumeratum_2.12-1.5.16-SNAPSHOT.pom
306[error]   not found: https://oss.sonatype.org/content/repositories/releases/com/beachape/enumeratum_2.12/1.5.16-SNAPSHOT/enumeratum_2.12-1.5.16-SNAPSHOT.pom
307[error]   not found: https://repo.typesafe.com/typesafe/releases/com/beachape/enumeratum_2.12/1.5.16-SNAPSHOT/enumeratum_2.12-1.5.16-SNAPSHOT.pom
308[error] Error downloading com.beachape:enumeratum-test_2.12:1.5.16-SNAPSHOT
309[error]   Not found
310[error]   Not found
311[error]   not found: /hom
```

Reply:
It doesn't. Are snapshots supposed to be published? Locally I have to do an incremental `publishLocal` like this:
```bash
sbt '+macros-aggregate/publishLocal; +core-aggregate/publishLocal; +test-aggregate/publishLocal; +test'
```
for it to work.

How does the publishing process usually work given this internal dependency structure?

Reply:
I'm not actually sure why this isn't working as-is since this PR isn't bumping any versions.  We have the stable version published to Maven Central, so it should just work 😕

Reply:
Assuming the issue is ScalaJS, I think if you add `dependsOn(core)` to these ScalaJS projects, it might fix the build.

Reply:
Yes that's the reason and that's why I've set the versions to head. I also read your comment this way:

> Thanks, leaving for a bit to see if CI will fix itself (you can go ahead and commit the references to use `.head` for this PR; I'll fix them later)

But I didn't realize that you can also just publish the current version for scalajs 1.0. So I've changed all references to `stable` now.

So now locally I can do a 
```bash
 sbt +macrosJS/publishLocal; +coreJS/publishLocal;  +enumeratumTestJS/publishLocal;
```
to just publish the scalajs 1.0 artifacts and then I'll be able to run the tests.